### PR TITLE
Fixes #360 Use auth middleware and move github requests

### LINKED ISSUE #360: App Crashes on performing authorised operations when not logged in
### This is a
- [x] Bug report
- [ ] Feature request

### Current behaviour

The application crashed when a user performs an action without logging in.
### Expected behaviour

The application shouldn't crash. Instead, the user should be redirected to homepage notifying the error

### Steps to reproduce
One of  the notable crashes
1. Login to yaydocCI
2. Clear browser cache
3. Perform repository delete operation

Here, the repository is deleted from the server but crashed when deleting hook due to the unavailability of user hook

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

* Moved all Github requests to `backend/github.js`
* Update by adding auth middleware in repository register and repository delete
## Related Issue

issue #360 
## How Has This Been Tested?

https://yaydoc7.herokuapp.com
## Types of changes

- [x] Bug fix 
- [ ] New feature
- [x] Code cleanup
## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y Your reviews?

File: backend/github.js [hunk 1]
```diff
@@ -96,3 +96,61 @@ exports.deleteHook = function (name, hook, accessToken, callback) {
...
+  }, callback);
+};
+
+/**
+ * Register a hook to the repository for Yaydoc
+ * @param name: `full_name` of the repository
+ * @param accessToken: Access token
+ * @param callback
+ */
+exports.registerHook = function (name, accessToken, callback) {
```
Comment (sch00lb0y) [inline backend/github.js]: please write this as Promise. because I already wrote the function as Promises in 
#318   remaining looks fine

Comment (imujjwal96) [inline backend/github.js]: Anything i write will be overwritten by your implementation. I don't think there is a need to update it now

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y  Updated and resolved conflicts. Please review!

**Comment by imujjwal96** [comment]
@sch00lb0y once more? :rofl:

**Pull request merged**